homeassistant/components/avri/sensor.py

```diff
@@ -1,68 +1,49 @@
 """Support for Avri waste curbside collection pickup."""
-from datetime import timedelta
 import logging
 
 from avri.api import Avri, AvriException
-import voluptuous as vol
 
-from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import CONF_NAME
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ID, DEVICE_CLASS_TIMESTAMP
 from homeassistant.exceptions import PlatformNotReady
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN, ICON
 
 _LOGGER = logging.getLogger(__name__)
-CONF_COUNTRY_CODE = "country_code"
-CONF_ZIP_CODE = "zip_code"
-CONF_HOUSE_NUMBER = "house_number"
-CONF_HOUSE_NUMBER_EXTENSION = "house_number_extension"
-DEFAULT_NAME = "avri"
-ICON = "mdi:trash-can-outline"
-SCAN_INTERVAL = timedelta(hours=4)
-DEFAULT_COUNTRY_CODE = "NL"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_ZIP_CODE): cv.string,
-        vol.Required(CONF_HOUSE_NUMBER): cv.positive_int,
-        vol.Optional(CONF_HOUSE_NUMBER_EXTENSION): cv.string,
-        vol.Optional(CONF_COUNTRY_CODE, default=DEFAULT_COUNTRY_CODE): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    }
-)
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
Optional: If you want to avoid the breaking change, you can import the existing config.  An example of how to do this is in `nut/sensor.py`.

Reply:
I'm not a big fan of backward compatability. Luckily the integration is niche enough that the amount of users affected will be very small. I'd rather keep the integration as clean as possible 😃.